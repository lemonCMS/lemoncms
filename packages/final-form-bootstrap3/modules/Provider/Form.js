import React from "react";
import PropTypes from "prop-types";
import { Form as FinalForm } from "react-final-form";
import arrayMutators from "final-form-arrays";
import ContextWrapper from "../Context/ContextWrapper";

const Form = props => {
  const { children, className, debug, readOnly, listen, ...formProps } = props;

  return (
    <FinalForm
      {...formProps}
      mutators={{
        ...arrayMutators
      }}
      render={({ handleSubmit, ...rest }) => {
        return (
          <ContextWrapper
            {...rest}
            debug={debug}
            listen={listen}
            readOnly={readOnly}
          >
            <form onSubmit={handleSubmit} className={className}>
              {children}
            </form>
          </ContextWrapper>
        );
      }}
    />
  );
};

Form.propTypes = {
  className: PropTypes.string,
  validate: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  debug: PropTypes.bool,
  readOnly: PropTypes.bool,
  listen: PropTypes.func,
  initialValues: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
      PropTypes.object,
      PropTypes.bool
    ])
  )
};

Form.defaultProps = {
  className: "",
  validate: () => {},
  listen: () => {},
  debug: false,
  readOnly: false,
  initialValues: {}
};

export default Form;
