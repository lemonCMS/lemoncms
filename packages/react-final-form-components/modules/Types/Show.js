import React from "react";
import PropTypes from "prop-types";
import context from "../decorators/consumer";

const Show = ({ condition, children, context: { checkCondition } }) => {
  if (condition && typeof condition === "function") {
    if (checkCondition(condition) !== true) {
      return null;
    }
    return children;
  }

  return null;
};

Show.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array,
    PropTypes.string
  ]).isRequired,
  condition: PropTypes.func
};
export default context(Show);
