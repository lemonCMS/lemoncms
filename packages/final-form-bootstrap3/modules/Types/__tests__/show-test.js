import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Form from "../../Provider/Form";
import Show from "../Show";
import React from "react";

describe("BS3 Text type", () => {
  it("should render input text", () => {
    const tree = renderer.create(
      <Form onSubmit={() => {}} initialValues={{ field: "true" }}>
        <Show name="show" condition={({ field }) => field === "true"}>
          SHOW
        </Show>
      </Form>
    );
    const output = tree.toJSON();
    expect(output).toMatchSnapshot();
  });

  it("should display SHOW", () => {
    const component = mount(
      <Form onSubmit={() => {}} initialValues={{ field: "true" }}>
        <Show name="show" condition={({ field }) => field === "true"}>
          SHOW
        </Show>
      </Form>
    );

    const div = component.find("div");
    expect(div).toBeDefined();
    expect(div.text()).toEqual("SHOW");
  });

  it("should not display SHOW", () => {
    const component = mount(
      <Form onSubmit={() => {}} initialValues={{ field: "false" }}>
        <Show name="show" condition={({ field }) => field === "true"}>
          SHOW
        </Show>
      </Form>
    );

    const div = component.find("div");
    expect(div).toBeDefined();
    expect(div.text()).toBeFalsy();
  });
});
