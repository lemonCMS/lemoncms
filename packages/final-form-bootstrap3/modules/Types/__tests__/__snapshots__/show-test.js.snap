// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BS3 Text type should render input text 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group has-feedback"
  >
    SHOW
  </div>
</form>
`;
